solver/llbsolver/ops/user_windows.go

```diff
@@ -0,0 +1,57 @@
+package ops
+
+import (
+	"context"
+
+	"github.com/docker/docker/pkg/idtools"
+	"github.com/moby/buildkit/executor"
+	"github.com/moby/buildkit/solver/llbsolver/file"
+	"github.com/moby/buildkit/solver/llbsolver/ops/fileoptypes"
+	"github.com/moby/buildkit/solver/pb"
+	"github.com/moby/buildkit/util/windows"
+	"github.com/pkg/errors"
+	copy "github.com/tonistiigi/fsutil/copy"
+)
+
+func getReadUserFn(exec executor.Executor) func(chopt *pb.ChownOpt, mu, mg fileoptypes.Mount) (*copy.User, error) {
+	return func(chopt *pb.ChownOpt, mu, mg fileoptypes.Mount) (*copy.User, error) {
+		return readUser(chopt, mu, mg, exec)
+	}
+}
+
+func readUser(chopt *pb.ChownOpt, mu, mg fileoptypes.Mount, exec executor.Executor) (*copy.User, error) {
```

Comment:
Shouldn't this function pass `snapshot.Mountable` instead of these multiple interface detections?

Reply:
This function was moved without change from its previous location. This function gets called in `solver/llbsolver/file/backend.go` where the `Mkdir()` function gets a `user` and `group` parameter which are `fileoptypes.Mount{}`.

We can convert it here or before we pass it in in the above mentioned package. I am ok with either version.

Reply:
Yes, but previously this was private behavior inside a package. Now there is a boundary between packages and looks like the types are not really correct but need multiple levels of detection on runtime and extra error validation to make sure the correct type was indeed passed.

Reply:
yup. Let me attempt a fix for that. Will push it along with the rest of the changes and discuss if it's ok or if a better approach is desired.